src/Compilers/VisualBasic/Portable/Symbols/Source/OverrideHidingHelper.vb

```diff
@@ -581,7 +581,9 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
 
         ' Comparer for comparing signatures of TSymbols in a runtime-equivalent way.
         ' It is not ReadOnly because it is initialized by a Shared Sub New of another instance of this class.
+#Disable Warning IDE0044 ' Add readonly modifier - Adding readonly generates compile error in the constructor.
         Private Shared s_runtimeSignatureComparer As IEqualityComparer(Of TSymbol)
+#Enable Warning IDE0044 ' Add readonly modifier
```

Comment:
Why ReadOnly isn't allowed here?

https://docs.microsoft.com/en-us/dotnet/visual-basic/language-reference/modifiers/readonly

> You can assign a value to a ReadOnly variable only in its declaration or in the constructor of a class or structure in which it is defined.

Is a static (Shared) constructor a special case that doesn't allow initialization of ReadOnly fields?

Reply:
Not sure, but we'll want to file a bug for this

Reply:
@sharwell I couldn't reproduce any of the problems with a small snippet to file a bug. :-(
Feeling like I'm missing something.